### ISSUE TITLE
Adding missing peripherals 

### ISSUE THREAD

**Issue by Triticum0** [xemu-project/xemu#389]
### Feature Request
There are a couple of missing peripherals which need to be implemented into Xemu to make some games become playable and would improve the emulator as a whole. Here is a full list of all the ones I know that need to be added.

- [ ] Gametrak
- [x] Xbox Communicator Set (#1126)
- [x] Xbox-Compatible Microphone
- [ ] MadCatz XBOX Steering Wheel
- [ ] Steel Battalion Controller
- [ ] Mad Catz Xbox Blaster
- [ ] Silent Scope Light Rifle
- [ ] Dance Dance Revolution Game Pad
- [x] Xbox Memory Card https://github.com/xemu-project/xemu/pull/1315

UI for above peripherals: #1128

### Games which become playable
Real World Golf
N.U.D.E.@: Natural Ultimate Digital Experiment
Silent Scope Collection
Steel Battalion
Karaoke Revolution
Karaoke Revolution Party
Dance Dance Revolution Ultramix Series  (Not required To play)
Dance UK
https://xemu.app/titles/53540001/#Zillernet (needs mic)

### Additional Context

I'm not sure if there other games that have been omitted by mistake, which are not on the list, and If any peripherals are already implemented, feel free to comment below to correct me.
Source:https://gamicus.fandom.com/wiki/List_of_Xbox_peripherals

**Comment by GXTX** [comment]
There's nothing stopping you from using these already. You plug them into your computer and tell xemu to use them via command line.

There's some extra steps you must go through to get it working on Windows (iirc) but Linux is straight forward.

**Comment by Triticum0** [comment]
Is there a guide on how to do this on windows and does this include Gametrak and Steel Battalion Controller because pcsx2 hasn't even implemented Gametrak
Also XMU hasn't been implemented https://github.com/mborgerson/xemu/issues/90

**Comment by GXTX** [comment]
Read the QEMU man pages on USB pass through, and XMU support is already there (it's just a generic storage device), just again not via the UI.

**Comment by mborgerson** [comment]
Pass-thru will work for some peripherals. For light guns it will probably not work, as they use they are expected to use the yellow wire, however these can be emulated.

xemu will eventually expose an interface for configuring XMUs+pass-thru+emulated peripherals. In the mean time, we consider the path "advanced" and don't really provide user support for it. That said, I'd accept a PR for the website which plainly describes how to do it for users who want to get their hands dirty.

**Comment by Triticum0** [comment]
Should #90 and this pr be closed then if most peripherals works?
should i tick all of them accept of the light gun?

**Comment by mborgerson** [comment]
No, we should not consider them complete until they are properly supported in the user interface.

**Comment by Triticum0** [comment]
Ok i will leave the pr as is

**Comment by HadetTheUndying** [comment]
> Pass-thru will work for some peripherals. For light guns it will probably not work, as they use they are expected to use the yellow wire, however these can be emulated.
> 
> xemu will eventually expose an interface for configuring XMUs+pass-thru+emulated peripherals. In the mean time, we consider the path "advanced" and don't really provide user support for it. That said, I'd accept a PR for the website which plainly describes how to do it for users who want to get their hands dirty.

How? I've poured over the qemu doc, and I am able to pass a device through via CLI arguments but xemu does not read any of of the devices as controllers.

I've tried
`./dist/xemu -devusb-host,vendorido0x045e,productid=0x8289D` for my controller S, I've tried this for my Steel Battalion Controller, DDR pad, lightgun, Madcatz Racing wheel, etc.
I've tried using the `port=x`  argument, the monitor output in `info usb` shows that it is creating a USB device but the device is never actually passed through.  Using `port=x.x` provides  `-device usb-host,vendorid=0x045e,productid=0x8289,port=3.1: usb port 3.1 (bus usb-bus.1) not found (in use?)` for any port numerical value

Doing any of these arguments in QEMU works fine for me for standard virtual machines, I have compiled with ` --enable-libusb --enable-usb-redir --enable-spice --enable-spice-protocol` I'm at a loss and I'd like to figure this out so I can have a more controlled environment to debug whatever is bottlenecking performance in Steel Battalion. 

I apologize for continually asking this question but I've been trying to get this working for 3 days now. Once again I've read the qemu doc and tried a ton of things like trying to assign a bus etc. 

Screenshot for reference that xemu is claiming it sees the host device 
![image](https://user-images.githubusercontent.com/28496334/169646668-5c7dd2a5-93a2-4dbe-98a8-940b6c1d0f5e.png)

**Comment by Florin9doi** [comment]
- [ ] Xbox Video Chat
![image](https://user-images.githubusercontent.com/2995486/174434700-3a21a843-feef-4e5f-bb6e-d49920e26f53.png)

**Comment by Darknior** [comment]
Hi,
The USB pass through is a very god idea for users that have the original material.
But for others, for exemple for gun games, it can help a lot pc gamers t make these games with mouse :)
if it was possible like on other emulator to emulate gun with mouse, we will can play these games with recent material, like some guns that working fine on CLD screens :)
Thanks

**Comment by GXTX** [comment]
> Hi, The USB pass through is a very god idea for users that have the original material. But for others, for exemple for gun games, it can help a lot pc gamers t make these games with mouse :) if it was possible like on other emulator to emulate gun with mouse, we will can play these games with recent material, like some guns that working fine on CLD screens :) Thanks

On my fork I have a branch that does mouse support, but it's not "finished".

**Comment by Darknior** [comment]
> On my fork I have a branch that does mouse support, but it's not "finished".

Excellent, so cool you have work on this feature. i have found it : https://github.com/GXTX/xemu
It is for an Batocera integration, on our system all gun games are ready to use with all guns or a wiimote.
You plug you play, nothing to do. But it is a hard and long work to release that ....
What is not finished on your fork ? Can we try gun games with mouse or not ?

**Comment by GXTX** [comment]
<https://github.com/GXTX/xemu/tree/mouse>

It's badly designed, replaces keyboard functionality, doesn't handle the corners correctly, captures mouse for the entire screen instead of just over the window. Lots of little stuff, and I don't really have much interest in finishing it. Works pretty well though. Played a bit of VCOP with it.

**Comment by Tovarichtch** [comment]
> https://github.com/GXTX/xemu/tree/mouse
> 
> It's badly designed, replaces keyboard functionality, doesn't handle the corners correctly, captures mouse for the entire screen instead of just over the window. Lots of little stuff, and I don't really have much interest in finishing it. Works pretty well though. Played a bit of VCOP with it.

Hi. Did you make any change in your fork since?

**Comment by WebSiteFurbo** [comment]
> https://github.com/GXTX/xemu/tree/mouse
> 
> It's badly designed, replaces keyboard functionality, doesn't handle the corners correctly, captures mouse for the entire screen instead of just over the window. Lots of little stuff, and I don't really have much interest in finishing it. Works pretty well though. Played a bit of VCOP with it.

Hi GXTX, 

is the first time I have written to you, and I would like to take this opportunity to thank you for creating Xemu. Like Tovarichtch, I would like to ask if there are any plans in the future to officially implement the use of the mouse as a lightgun in Xemu. I would be willing to fund this change through Patreon or another platform, should you enable paid requests. 

Kind regards and keep up the good work.
